sdk/identity/azure_identity/src/credentials/client_assertion_credentials.rs

```diff
@@ -68,59 +80,92 @@ impl<C: ClientAssertion> ClientAssertionCredential<C> {
         assertion: C,
         options: Option<ClientAssertionCredentialOptions>,
     ) -> azure_core::Result<Self> {
+        validate_tenant_id(&tenant_id)?;
+        validate_not_empty(&client_id, "no client ID specified")?;
+        let options = options.unwrap_or_default().credential_options;
+        let endpoint = options
+            .authority_host()?
+            .join(&format!("/{tenant_id}/oauth2/v2.0/token"))
+            .with_context(ErrorKind::DataConversion, || {
+                format!("tenant_id '{tenant_id}' could not be URL encoded")
+            })?;
         Ok(Self {
-            tenant_id,
             client_id,
             assertion,
+            endpoint,
             cache: TokenCache::new(),
-            options: options.unwrap_or_default(),
+            options,
         })
     }
 
-    async fn get_token(&self, scopes: &[&str]) -> azure_core::Result<AccessToken> {
-        let token = self.assertion.secret().await?;
-        let credential_options = &self.options.credential_options;
-        let res: AccessToken = federated_credentials_flow::authorize(
-            credential_options.http_client().clone(),
-            &self.client_id,
-            &token,
-            scopes,
-            &self.tenant_id,
-            &credential_options.authority_host()?,
-        )
-        .await
-        .map(|r| {
-            AccessToken::new(
-                r.access_token().clone(),
-                OffsetDateTime::now_utc() + Duration::from_secs(r.expires_in),
-            )
-        })
-        .context(ErrorKind::Credential, "request token error")?;
-        Ok(res)
+    async fn get_token_impl(&self, scopes: &[&str]) -> azure_core::Result<AccessToken> {
+        let mut req = Request::new(self.endpoint.clone(), Method::Post);
+        req.insert_header(
+            headers::CONTENT_TYPE,
+            content_type::APPLICATION_X_WWW_FORM_URLENCODED,
+        );
+        let assertion = self.assertion.secret().await?;
+        let encoded: String = form_urlencoded::Serializer::new(String::new())
```

Comment:
Ah, you import it for url-encoded bodies. Maybe we should re-export it from `azure_core` as well. For URIs you can actually get a mutable one from `Url`, but that's clearly not what you need here.

Reply:
Maybe. I suppose it depends on the problem you hope to solve by re-exporting. I doubt the export would often be used, because I don't find many form bodies in the Go repo and eventually this crate won't craft requests

Reply:
I suppose we don't need to make this generally available to crate devs - only Identity - so probably don't re-export then. The idea is that we want to limit Azure SDK crate devs to types on `azure_core` - even hiding anything "typespec" in general.